app/src/main/java/edu/wisc/ece/pinpoint/MainActivity.java

```diff
@@ -44,6 +47,15 @@ public void onCreate(Bundle savedInstanceState) {
                 mapButton.setVisibility(View.VISIBLE);
             }
         });
+
+        // Fetch dropped & found pins for cached access later
+        FirebaseDriver firebase = FirebaseDriver.getInstance();
```

Comment:
I think since we are using Tasks the main thread will continue executing while the async request completes and the listeners will only get called once the operation is done.